drivers/sensor/hx711/hx711.c

```diff
@@ -0,0 +1,562 @@
+			    * (data->reading - data->offset);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+/**
+ * @brief Initialise HX711.
+ *
+ * @param dev Pointer to the hx711 device structure
+ *
+ * @retval 0 on success
+ * @retval -EINVAL if an invalid argument is given
+ *
+ */
+static int hx711_init(const struct device *dev)
+{
+	LOG_DBG("Initialising HX711\n");
+
+	int ret = 0;
+	struct hx711_data *data = dev->data;
+	const struct hx711_config *cfg = dev->config;
+
+	LOG_DBG("SCK GPIO port : %s\n", cfg->sck_ctrl);
+	LOG_DBG("SCK Pin : %d\n", cfg->sck_pin);
+	LOG_DBG("DOUT GPIO port : %s\n", cfg->dout_ctrl);
+	LOG_DBG("DOUT Pin : %d\n", cfg->dout_pin);
+	LOG_DBG("RATE GPIO port : %s\n", cfg->rate_ctrl);
+	LOG_DBG("RATE Pin : %d\n", cfg->rate_pin);
+	LOG_DBG("Gain : %d\n", data->gain);
+	LOG_DBG("Offset : %d\n", data->offset);
+	LOG_DBG("Slope : %d.%d\n", data->slope.val1, data->slope.val2);
+
+	/* Configure SCK as output, LOW */
+	data->sck_gpio = device_get_binding(cfg->sck_ctrl);
+	if (data->sck_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->sck_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("SCK pin controller is %p, name is %s\n",
+		data->sck_gpio,
+		data->sck_gpio->name);
+
+	ret = gpio_pin_configure(data->sck_gpio,
+				 cfg->sck_pin,
+				 GPIO_OUTPUT_INACTIVE | cfg->sck_flags);
+	if (ret != 0) {
+		return ret;
+	}
+
+#if DT_INST_NODE_HAS_PROP(0, rate_gpios)
+	/* Configure RATE as output, LOW */
+	data->rate_gpio = device_get_binding(cfg->rate_ctrl);
+	if (data->rate_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->rate_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("RATE pin controller is %p, name is %s\n",
+		data->rate_gpio,
+		data->rate_gpio->name);
+	ret = gpio_pin_configure(data->rate_gpio,
+				 cfg->rate_pin,
+				 GPIO_OUTPUT_INACTIVE | cfg->rate_flags);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_set(data->rate_gpio,
+			   hx711_config.rate_pin,
+			   CONFIG_HX711_SAMPLING_RATE);
+	if (ret != 0) {
+		return ret;
+	}
+#endif
+
+	/* Configure DOUT as input */
+	data->dout_gpio = device_get_binding(cfg->dout_ctrl);
+	if (data->dout_gpio == NULL) {
+		LOG_ERR("Failed to get GPIO device %s.", cfg->dout_ctrl);
+		return -EINVAL;
+	}
+	LOG_DBG("DOUT pin controller is %p, name is %s\n",
+		data->sck_gpio,
+		data->sck_gpio->name);
+	ret = gpio_pin_configure(data->sck_gpio,
+				 cfg->dout_pin,
+				 GPIO_INPUT | cfg->dout_flags);
+	if (ret != 0) {
+		return ret;
+	}
+	LOG_DBG("Set DOUT pin : %d\n", cfg->dout_pin);
+
+	/* Get a reading to set GAIN */
+	ret = hx711_sample_fetch(dev, SENSOR_CHAN_WEIGHT);
+
+	return ret;
+}
+
+/**
+ * @brief Zero the HX711.
+ *
+ * @param dev Pointer to the hx711 device structure
+ * @param readings Number of readings to get average offset.
+ *        5~10 readings should be enough, although more are allowed.
+ * @retval The offset value
+ *
+ */
+int avia_hx711_tare(const struct device *dev, uint8_t readings)
+{
+	int32_t avg = 0;
+	struct hx711_data *data = dev->data;
+
```

Comment:
Isn't it better to check readings for zero here and exit. There isn't much point to go further in such case.

Reply:
The reading is rarely zero. 
something like one in a million :)

Reply:
Won't fix.